### PR TITLE
SAAS-2597: GCP VPC Network Firewall Logging Metadata

### PR DESCRIPTION
**Pull request by mehakseedat63**

### CODE REVIEW

File: plugins/google/vpcnetwork/firewallLoggingMetadata.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
+var async = require('async');
+var helpers = require('../../../helpers/google');
+
+module.exports = {
+    title: 'Firewall Logging Metadata',
+    category: 'VPC Network',
+    description: 'Ensure that VPC Network firewall logging is configured to exclude logging metadata in order to reduce the size of the log files.',
+    more_info: 'You can significantly reduce the size of your log files and optimize storage costs by not including metadata, By default, metadata is included in firewall rule log files.',
```
Comment (AkhtarAmir) [inline plugins/google/vpcnetwork/firewallLoggingMetadata.js:8]:
```suggestion
    more_info: 'You can significantly reduce the size of your log files and optimize storage costs by not including metadata. By default, metadata is included in firewall rule log files.',
```

File: plugins/google/vpcnetwork/firewallLoggingMetadata.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+            let firewalls = helpers.addSource(
+                cache, source, ['firewalls', 'list', region]);
+
+            if (!firewalls) return rcb();
+
+            if (firewalls.err || !firewalls.data) {
+                helpers.addResult(results, 3, 'Unable to query firewall rules', region, null, null, firewalls.err);
+                return rcb();
+            }
```
Comment (AkhtarAmir) [inline plugins/google/vpcnetwork/firewallLoggingMetadata.js:53]:
```suggestion
            if (!firewalls || firewalls.err || !firewalls.data) {
                helpers.addResult(results, 3, 'Unable to query firewall rules', region, null, null, firewalls.err);
                return rcb();
            }
```

File: plugins/google/vpcnetwork/firewallLoggingMetadata.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+            if (!firewalls.data.length) {
+                helpers.addResult(results, 0, 'No firewall rules found', region);
+                return rcb();
+            }
+
+            var loggedMetadataVPCs = [];
+            firewalls.data.forEach(firewall => {
+                if (!firewall.disabled && firewall.logConfig && firewall.logConfig.enable && firewall.logConfig.metadata == 'INCLUDE_ALL_METADATA') {
+                    loggedMetadataVPCs.push(firewall.network);
```
Comment (AkhtarAmir) [inline plugins/google/vpcnetwork/firewallLoggingMetadata.js:63]:
```suggestion
                if (!firewall.disabled && firewall.logConfig && firewall.logConfig.enable && firewall.logConfig.metadata == 'INCLUDE_ALL_METADATA' && firewall.network && !loggedMetadataVPCs.includes(firewall.network)) {
                    loggedMetadataVPCs.push(firewall.network);
```

File: plugins/google/vpcnetwork/firewallLoggingMetadata.js [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+                }   
+            });
+
+            networks.data.forEach(network => {
+                if (!network.name) return;
+
+                let resource = helpers.createResourceName('networks', network.name, project, 'global');
+                if (loggedMetadataVPCs.includes(network.selfLink)){
```
Comment (AkhtarAmir) [inline plugins/google/vpcnetwork/firewallLoggingMetadata.js:71]:
```suggestion
                if (network.selfLink && loggedMetadataVPCs.includes(network.selfLink)){
```

File: plugins/google/vpcnetwork/firewallLoggingMetadata.js [hunk 5]
```diff
@@ -0,0 +1,86 @@
+var async = require('async');
+var helpers = require('../../../helpers/google');
+
+module.exports = {
+    title: 'Firewall Logging Metadata',
+    category: 'VPC Network',
+    description: 'Ensure that VPC Network firewall logging is configured to exclude logging metadata in order to reduce the size of the log files.',
+    more_info: 'You can significantly reduce the size of your log files and optimize storage costs by not including metadata, By default, metadata is included in firewall rule log files.',
+    link: 'https://cloud.google.com/vpc/docs/firewall-rules-logging',
+    recommended_action: 'Create Cloud DNS Server Policy with logging enabled for VPC Networks',
```
Comment (AkhtarAmir) [inline plugins/google/vpcnetwork/firewallLoggingMetadata.js:10]: This doesn't seem right

**Review by AkhtarAmir — CHANGES_REQUESTED** [review]

**Review by AkhtarAmir — APPROVED** [review]

**Review by dipsubha06 — APPROVED** [review]

**Pull request merged**